partiql-eval/src/main/java/org/partiql/eval/StructValue.java

```diff
@@ -0,0 +1,37 @@
+package org.partiql.eval;
+
+import org.jetbrains.annotations.NotNull;
+import org.partiql.value.PartiQLValueType;
+
+import java.util.Collections;
+import java.util.Iterator;
+
+/**
+ * This shall always be package-private (internal).
+ */
+class StructValue implements PQLValue {
+
+    @NotNull
+    final Iterable<StructField> _value;
+
+    StructValue(@NotNull Iterable<StructField> value) {
+        _value = value;
+    }
+
+    @Override
+    public boolean isNull() {
+        return false;
+    }
+
+    @Override
+    @NotNull
+    public Iterator<StructField> getStructFields() {
+        return _value.iterator();
+    }
+
+    @NotNull
+    @Override
+    public PartiQLValueType getType() {
+        return PartiQLValueType.STRUCT;
+    }
+}
```

Comment:
We should have a map and multi-map implementation. When a "partiql struct" does not have duplicate keys, then we use the map which is key->value, but when there are duplicates, we store linked-lists as the map entries.

So we have `key->list<field>` where the field is as earlier which preserves the original field name case. Keys are all case-normalized